packages/sd-jwt/src/repository/SdJwtRecord.ts

```diff
@@ -0,0 +1,45 @@
+import type { TagsBase } from '@aries-framework/core'
+
+import { BaseRecord, utils } from '@aries-framework/core'
+
+export type SdJwtRecordTags = TagsBase
+
+export type SdJwtRecordStorageProps = {
+  id?: string
+  createdAt?: Date
+  tags?: SdJwtRecordTags
+  sdJwt: string
+}
+
+export type SaveSdJwtRecordOptions = {
+  id?: string
+  sdJwt: string
+  tags?: SdJwtRecordTags
+}
+
+export class SdJwtRecord extends BaseRecord<SdJwtRecordTags> {
+  /*
+   * @todo storing compact is not permanent
+   */
+  public sdJwt!: string
+
+  public static readonly type = 'SdJwtRecord'
+  public readonly type = SdJwtRecord.type
+
+  public constructor(props: SdJwtRecordStorageProps) {
+    super()
+
+    if (props) {
+      this.id = props.id ?? utils.uuid()
+      this.createdAt = props.createdAt ?? new Date()
+      this.sdJwt = props.sdJwt
+      this._tags = props.tags ?? {}
+    }
+  }
+
+  public getTags() {
```

Comment:
No tags at all will make it especially hard to query the credentials for input in a VP...

Reply:
yeah I was wondering what the best tag structure would be for this. Right now I added the disclosure keys as an array, but I am not sure whether that is supported within askar.


Will likely also include the claimKeys, combined with the selectively disclosable keys.

Reply:
You can add arrays. It will currently be added as an `<attr>:<value>: 1/0` I think. So `disclosureKeys:age: 1`. Maybe askar also supports something for this natively, which we maybe should look into

Reply:
Tags will be last as-is for now. Once we do the integration with openid they can be added as at that point we will know exactly how to query.